jsonargparse/_typehints.py

```diff
@@ -919,7 +919,11 @@ def adapt_typehints(
     elif typehint_origin in mapping_origin_types:
         if isinstance(val, NestedArg):
             if isinstance(prev_val, dict):
-                val = {**prev_val, val.key: val.val}
+                if isinstance(val.key, str) and "." in val.key:
+                    key_prefix, key_suffix = val.key.split(".", 1)
+                    val = {**prev_val, key_prefix: {key_suffix: val.val}}
```

Comment:
These two lines should be covered with the new test. You could run
```
pytest --cov --cov-report=html jsonargparse_tests/test_typehints.py::test_<new_function_name>
```
And open `htmlcov/index.html` to figure out if these lines are covered. Or you could run the test with debugging and set a breakpoint here to be sure that it is working as expected.